irods/manager/data_object_manager.py

```diff
@@ -296,7 +323,8 @@ def open_with_FileRaw(self, *arg, **kw_options):
                                kw.RESC_HIER_STR_KW
                            ))
 
-    def open(self, path, mode, create = True, finalize_on_close = True, **options):
+    def open(self, path, mode, create = True, finalize_on_close = True, auto_close = client_config.getter('data_objects','auto_close'),
```

Comment:
My understanding was that it's best to follow a pattern like this:
```suggestion
    def open(self, path, mode, create = True, finalize_on_close = True, auto_close = None,
```
And then you can call `client_config.getter('data_objects','auto_close')` in the function if `auto_close` is `None`. This prevents `client_config.getter('data_objects','auto_close')` from being called on import(?). Maybe that's not right...

Reply:
`getter` is most likely called from this context when the open function gets compiled, ie at the time that the data_object_manager module is imported.
The lambda returned by getter is called in the function.
Because `auto_close` parameter can, and does, take on non-callable values, we've deferred the calling of the thing until the function body, and then only do the call if that parameter holds a callable object.  That's how the defaulted parameter can reflect the current global `auto_close` setting...

Reply:
ok, commented the parameter ....

Reply:
Great. Please resolve this conversation. Thanks